core/state_transition.go

```diff
@@ -249,15 +246,15 @@ func (st *StateTransition) TransitionDb() (ret []byte, requiredGas, usedGas *big
 		// sufficient balance to make the transfer happen. The first
 		// balance transfer may never fail.
 		if vmerr == vm.ErrInsufficientBalance {
-			return nil, nil, nil, false, vmerr
+			return nil, nil, nil, vmerr, vmerr
```

Comment:
Here you should have returned `nil`, not `vmerr` for the changed element.

Reply:
Yeah I was torn on that one.  Reason that I ended up sending it the vmerr field is that it *is* a VM error.  Reason that I left it in the err field is that it retains compatibility with existing code that expects it to show up there.  I figured that this was the best of the options, although I'm not married to it.